.github/workflows/build.yml

```diff
@@ -54,6 +54,86 @@ jobs:
         run: |
           make format || true
           make test
+  test-linux-s390x:
+    if: (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release*')
+    runs-on: ubuntu-latest
+    env:
+      ZVSI_FP_NAME: bp-floating-ci-${{ github.run_id }} 
+      ZVSI_INSTANCE_NAME: bp-zvsi-ci-${{ github.run_id }} 
+      ZVSI_VPC: r038-b86410fb-6889-4094-821d-1ceb72176bc4
+      ZVSI_ZONE_NAME: ca-tor-1
+      ZVSI_PROFILE_NAME: bz2-4x16
+      ZVSI_SUBNET: 02q7-af49cf76-0471-49f4-a245-b60d63196efb
+      ZVSI_IMAGE: r038-67da7290-933e-42e4-a1c2-ff439b4b6d2c
+      ZVSI_KEY: r038-f8e35cee-0475-4315-a051-40f4d2fb6af9
+      ZVSI_RG_ID: 31e0f1b03d1b4d42bd98b6768ba5421a
+      ZVSI_SG: r038-1e644f46-8044-4043-9108-2b7dc178827f
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - name: install ibmcli and setup ibm login 
+        run: |
+          curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+          ibmcloud login -q --apikey ${{ secrets.IBMCLOUD_API_KEY }} -r ca-tor
+          ibmcloud plugin install vpc-infrastructure 
+      - name: Creation of ZVSI
+        id: ZVSI
+        run: |
+          #creation of zvsi
+          ibmcloud is instance-create $ZVSI_INSTANCE_NAME $ZVSI_VPC $ZVSI_ZONE_NAME $ZVSI_PROFILE_NAME $ZVSI_SUBNET --image $ZVSI_IMAGE --keys $ZVSI_KEY --resource-group-id $ZVSI_RG_ID --sgs $ZVSI_SG
+          #Reserving a floating ip to the ZVSI
+          ibmcloud is floating-ip-reserve $ZVSI_FP_NAME --zone $ZVSI_ZONE_NAME --resource-group-id $ZVSI_RG_ID --in $ZVSI_INSTANCE_NAME
+          #Bouding the Floating ip to the ZVSI
+          ibmcloud is floating-ip-update $ZVSI_FP_NAME --nic primary --in $ZVSI_INSTANCE_NAME          
+          sleep 60
+          #Saving the Floating IP to login ZVSI
+          ZVSI_HOST=$(ibmcloud is floating-ip $ZVSI_FP_NAME | awk '/Address/{print $2}')
+          echo $ZVSI_HOST
+          echo "IP=${ZVSI_HOST}" >> $GITHUB_OUTPUT
+      - name: Status of ZVSI
+        run: |
+          check=$(ibmcloud is ins| awk '/'$ZVSI_INSTANCE_NAME'/{print $3}')
+          while [[ $check != "running" ]]
+          do
+            check=$(ibmcloud is ins | awk '/'$ZVSI_INSTANCE_NAME'/{print $3}')
+            if [[ $check == 'failed' ]]
+            then
+              echo "Failed to run the ZVSI"
+              break
+              fi
+          done
+      - name: Install dependencies and run all tests on s390x ZVSI
+        uses: appleboy/ssh-action@v0.1.10
```

Comment:
I don't think we have any set criteria, but let me ask around to see if there are guidelines we can follow

Reply:
I didn't find much beyond the guidance [here](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) ...but I think we could mitigate the risk by moving this job to its own workflow file. That way, any secrets not used by the workflow would be scrubbed according to [here](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#accessing-secrets). We could also [restrict the permissions](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes) on the `GITHUB_TOKEN` for the new workflow to be read-only. 

@dilipgb would you be amenable to making these changes to improve our security posture? I'm happy to work with you on it if that would be helpful. I don't think this requires further end-to-end testing on a fork, as building and storing the artifacts for s390x can still happen within build.yml since we are using cross-compilation.

Reply:
@natalieparellano I seperated s390x related tests to separate workflow. Can you please take a look?